tests/gis_tests/test_geoip.py

```diff
@@ -32,6 +34,23 @@ class GeoIPTest(unittest.TestCase):
     addr = '128.249.1.1'
     fqdn = 'tmc.edu'
 
+    def _is_dns_available(self, domain):
+        # Naive check to see if there is DNS available to use.
+        # Used to conditionally skip fqdn geoip checks.
+        # See #25407 for details.
+        if sys.version_info >= (3, 3):
+            # socket.error is deprecated since 3.3 and it is not an
+            # ancestor for other possible network exceptions (socket.gaierror,
+            # socket.timeout) anymore. Need to catch OSError instead.
+            ErrClass = OSError
+        else:
```

Comment:
Django 1.9 only supports Python 2.7 and 3.4+ so something like `ErrClass = socket.error if six.PY2 else OSError` should be okay.